frontend/src/components/assessments/student-experience/SubmitButton.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useContext, useEffect, useMemo } from "react";
+import { useMutation } from "@apollo/client";
+import { Button } from "@chakra-ui/react";
+
+import { SUBMIT_TEST } from "../../../APIClients/mutations/TestSessionMutations";
+import AuthContext from "../../../contexts/AuthContext";
+import StudentContext from "../../../contexts/StudentContext";
+import WriteAssessmentContext from "../../../contexts/WriteAssessmentContext";
+import {
+  isCompleted,
+  mapAnswersToResultsArray,
+} from "../../../utils/StudentUtils";
+import Modal from "../../common/Modal";
+import Toast from "../../common/Toast";
+
+const SubmitButton = (): React.ReactElement => {
+  const { testSession } = useContext(StudentContext);
+  const { answers, setIsSubmitted, setIsLoading } = useContext(
+    WriteAssessmentContext,
+  );
+  const { showToast } = Toast();
+  const { authenticatedUser } = useContext(AuthContext);
+  const [showConfirmModal, setShowConfirmModal] = React.useState(false);
+
+  const [submitTest, { loading }] = useMutation<{
+    submitTest: string;
+  }>(SUBMIT_TEST);
+
+  useEffect(() => {
+    if (loading) {
+      setIsLoading(true);
+    }
+  }, [loading, setIsLoading]);
+
+  const handleSubmitTest = async () => {
+    await submitTest({
+      variables: {
+        id: testSession?.id,
+        result: {
+          student: authenticatedUser?.id,
+          answers: mapAnswersToResultsArray(answers),
+        },
+      },
+    })
+      .then(() => {
+        setIsLoading(false);
+        setIsSubmitted(true);
+      })
+      .catch(() => {
+        setIsLoading(false);
+        showToast({
+          message: "Assessment failed to submit. Please try again.",
+          status: "error",
+        });
+      });
```

Comment:
nit: You can probably make use of async/await instead of then/catch
```suggestion
    try {
      await submitTest({
        variables: {
          id: testSession?.id,
          result: {
            student: authenticatedUser?.id,
            answers: mapAnswersToResultsArray(answers),
          },
        },
      });

      setIsSubmitted(true);
    } catch (e) {
      showToast({
        message: "Assessment failed to submit. Please try again.",
        status: "error",
      });
    }

    setIsLoading(false);
```